AK/Time.h

```diff
@@ -213,6 +213,16 @@ class Duration {
 
 public:
     [[nodiscard]] constexpr static Duration from_seconds(i64 seconds) { return Duration(seconds, 0); }
+    [[nodiscard]] constexpr static Duration from_seconds_f64(f64 seconds)
+    {
+        seconds = AK::max(seconds, static_cast<f64>(NumericLimits<i64>::min()));
+        seconds = AK::min(seconds, static_cast<f64>(NumericLimits<i64>::max()));
+        i64 integer_seconds = static_cast<i64>(seconds);
+        if (static_cast<f64>(integer_seconds) > seconds)
+            integer_seconds--;
+        u32 integer_nanoseconds = static_cast<u32>((seconds - static_cast<f64>(integer_seconds)) * 1'000'000'000);
```

Comment:
There's a lot of potential issues with the conversions happening here, e.g. NaNs get through or when `seconds` is large enough, it might not be enough to just do `integer_seconds--` since the difference with `seconds` might be larger than 1. This also could cause a negative value for `seconds - static_cast<f64>(integer_seconds)` resulting in an underflow when casting to `u32`. You might want to differentiate between the scenarios where the integral part of `seconds` is fully represented and where floating point imprecisions might turn up.

At any rate: please add a test case for this method so it's easier to expose and fix bugs in the future :)